src/System.Diagnostics.TraceSource/tests/TraceClassTests.cs

```diff
@@ -341,9 +351,42 @@ public void TraceTest01()
         }
 
         [Fact]
```

Comment:
I just chatted with @safern offline and AppContext.TargetFrameworkName is only available in .NET Core and .NET Framework 4.6.3/4.7 so it isn't useful for our purposes just yet. We should consider adding something similar with the proposal to update RuntimeInformation at https://github.com/dotnet/corefx/issues/16629